From b35c355d3bd6acc63d54237df4a0f1cef27e5bb3 Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Tue, 21 Oct 2025 13:26:22 -0700
Subject: [PATCH 477/553] tst_QList: call the two-arg qHash() function

The single-arg one is deprecated. Let's stop using them in tests.

Drive-by test that hashes work for zero and non-zero seeds.

Pick-to: 6.8 6.5
Change-Id: I5b32acf0797e9e62a5b7fffd1bc44f5372a044cf
Reviewed-by: Fabian Kosmale <fabian.kosmale@qt.io>
(cherry picked from commit 1b1b58ff17b1d8f3941f12f9f2213a6ada24f927)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/tools/qhashfunctions.h           | 4 ++--
 tests/auto/corelib/tools/qlist/tst_qlist.cpp | 9 +++++++--
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/corelib/tools/qhashfunctions.h b/src/corelib/tools/qhashfunctions.h
index 3e44f5f3442..68b0db30791 100644
--- a/src/corelib/tools/qhashfunctions.h
+++ b/src/corelib/tools/qhashfunctions.h
@@ -413,14 +413,14 @@ qHashMultiCommutative(size_t seed, const T &... args)
 
 template <typename InputIterator>
 inline size_t qHashRange(InputIterator first, InputIterator last, size_t seed = 0)
-    noexcept(noexcept(qHash(*first))) // assume iterator operations don't throw
+    noexcept(noexcept(qHash(*first, 0))) // assume iterator operations don't throw
 {
     return std::accumulate(first, last, seed, QtPrivate::QHashCombine(seed));
 }
 
 template <typename InputIterator>
 inline size_t qHashRangeCommutative(InputIterator first, InputIterator last, size_t seed = 0)
-    noexcept(noexcept(qHash(*first))) // assume iterator operations don't throw
+    noexcept(noexcept(qHash(*first, 0))) // assume iterator operations don't throw
 {
     return std::accumulate(first, last, seed, QtPrivate::QHashCombineCommutative(seed));
 }
diff --git a/tests/auto/corelib/tools/qlist/tst_qlist.cpp b/tests/auto/corelib/tools/qlist/tst_qlist.cpp
index e84c35691da..9a9d6fee474 100644
--- a/tests/auto/corelib/tools/qlist/tst_qlist.cpp
+++ b/tests/auto/corelib/tools/qlist/tst_qlist.cpp
@@ -2108,10 +2108,15 @@ void tst_QList::qhash() const
     TST_QLIST_CHECK_LEAKS(T)
 
     QList<T> l1, l2;
-    QCOMPARE(qHash(l1), qHash(l2));
+    QCOMPARE(qHash(l1, 0), qHash(l2, 0));
     l1 << SimpleValue<T>::at(0);
     l2 << SimpleValue<T>::at(0);
-    QCOMPARE(qHash(l1), qHash(l2));
+    QCOMPARE(qHash(l1, 0), qHash(l2, 0));
+
+    QCOMPARE(qHash(l1, 1), qHash(l2, 1));
+    l1.clear();
+    l2.clear();
+    QCOMPARE(qHash(l1, 1), qHash(l2, 1));
 }
 
 template <typename T>
-- 
2.51.2

